make/hotspot/lib/CompileJvm.gmk

```diff
@@ -83,26 +83,22 @@ CFLAGS_VM_VERSION := \
 ################################################################################
 # Disabled warnings
 
-DISABLED_WARNINGS_gcc := parentheses comment unknown-pragmas address \
-    delete-non-virtual-dtor char-subscripts array-bounds int-in-bool-context \
-    ignored-qualifiers  missing-field-initializers implicit-fallthrough \
-    empty-body strict-overflow sequence-point maybe-uninitialized \
-    misleading-indentation cast-function-type shift-negative-value
-
-ifeq ($(call check-jvm-feature, zero), true)
-  DISABLED_WARNINGS_gcc += return-type switch clobbered
-endif
+DISABLED_WARNINGS_gcc := array-bounds comment delete-non-virtual-dtor \
+    empty-body ignored-qualifiers implicit-fallthrough int-in-bool-context \
+    maybe-uninitialized missing-field-initializers parentheses \
+    shift-negative-value unknown-pragmas
+
+DISABLED_WARNINGS_clang := ignored-qualifiers sometimes-uninitialized \
+    missing-braces delete-non-abstract-non-virtual-dtor unknown-pragmas
 
-DISABLED_WARNINGS_clang := tautological-compare \
-    undefined-var-template sometimes-uninitialized unknown-pragmas \
-    delete-non-virtual-dtor missing-braces char-subscripts \
-    ignored-qualifiers missing-field-initializers mismatched-tags \
-    shift-negative-value misleading-indentation
+ifneq ($(DEBUG_LEVEL), release)
+  # Assert macro gives warning
+  DISABLED_WARNINGS_clang += tautological-constant-out-of-range-compare
+endif
 
 DISABLED_WARNINGS_xlc := tautological-compare shift-negative-value
 
-DISABLED_WARNINGS_microsoft := 4100 4127 4146 4201 4244 4291 4351 \
-    4511 4512 4514 4624
+DISABLED_WARNINGS_microsoft := 4624 4244 4291 4146
```

Comment:
I went through the list of Visual Studio warnings being removed from the
global suppression list.

clang: tautological-compare
windows: 4127 - conditional expression is constant
I'm not sure we want these turned on.  This is the kind of thing that can
happen quite easily with macros and/or in a code base that uses things like
conditional compilation to support multiple platforms.  Working around these
warnings when they arise can make for significant code uglification.  OTOH, I
guess it *could* catch some programming mistakes.  Maybe these could remain
surppressed for this change, and discussed separately?

4100 - 'identifier': unreferenced formal parameter
I'm very surprised removing this doesn't cause problems, as I'm sure I've seen
named but unused parameters. Maybe they've been cleaned up over time? I think
some compiler (probably Visual Studio) used to be unhappy with unnamed
parameters, which made this hard to avoid in some cases. Fortunately, that
doesn't seem to be a problem anymore.
Removal okay.

4201 - nonstandard extension used: nameless struct/union 
Removal okay. 

4351  - new behavior: elements of array ‘array’ will be default initialized (according to google, but it is undocumented!!!) 
This one seems to have been dropped by more recent versions of VS.
There are places where we use the feature, expecting the new behavior.
Removal okay. 

4511 - 'class': copy constructor was implicitly defined as deleted
4512 - 'class': assignment operator was implicitly defined as deleted  
I think that if the indicated situations were to exist and the warnings hadn't
been suppressed that we'd get link-time errors instead.
Removal okay. 

4514 - 'function': unreferenced inline function has been removed
I think we don't want this turned on, but it's also off by default.
Removal okay.